src/Http/Authentication.Abstractions/src/AuthenticationSchemeBuilder.cs

```diff
@@ -27,17 +27,25 @@ public AuthenticationSchemeBuilder(string name)
         /// <summary>
         /// The display name for the scheme being built.
         /// </summary>
-        public string DisplayName { get; set; }
+        public string? DisplayName { get; set; }
 
         /// <summary>
         /// The <see cref="IAuthenticationHandler"/> type responsible for this scheme.
         /// </summary>
-        public Type HandlerType { get; set; }
+        public Type? HandlerType { get; set; }
 
         /// <summary>
         /// Builds the <see cref="AuthenticationScheme"/> instance.
         /// </summary>
         /// <returns></returns>
-        public AuthenticationScheme Build() => new AuthenticationScheme(Name, DisplayName, HandlerType);
+        public AuthenticationScheme Build()
+        {
+            if (HandlerType is null)
+            {
+                throw new InvalidOperationException($"{nameof(HandlerType)} must be configured to build an {nameof(AuthenticationScheme)}.");
```

Comment:
I think we lost the type check for IAuthenticationHandler, that was removed from the scheme ctor

```
            if (!typeof(IAuthenticationHandler).IsAssignableFrom(handlerType))
            {	            {
                throw new ArgumentException("handlerType must implement IAuthenticationHandler.");	                throw new ArgumentException("handlerType must implement IAuthenticationHandler.");
            }
```

Reply:
That happens as part of the `AuthenticationScheme`: https://github.com/dotnet/aspnetcore/pull/22541/files#diff-5a26792de4ce3d4e74ffdf686c7e9ec1L31.

Reply:
Right sorry I got confused looking at why this code added here, I thought it got removed, so why do we need to check for null on HandlerType here as well in the ctor

Reply:
The ctor no longer allows null-values (it previously checked for nulls) as part of the ctor which continues to stay. Having the annotation makes that more explicit. However the builder now has the choice of using the forgiveness operator or doing something explicit when it encounters a null value. I prefer the latter, but I'll leave it up to you.

Reply:
I guess my only questions is whether this means we can remove the handlerType == null check from the ctor of AuthenticationScheme?

Reply:
That's still constructible by user code. It doesn't really hurt us to have the check in there